### PR TITLE
New Subworkflow: mapseq_otu_krona

### PR DESCRIPTION
**Pull request by chrisAta**
Subworkflow for generating taxonomic assignments for reads and visualisations with Krona

### CODE REVIEW

File: subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    db_fasta = ch_dbs[0]
+    db_tax = ch_dbs[1]
+    db_otu = ch_dbs[2]
+    db_mscluster = ch_dbs[3]
+    db_label = ch_dbs[4]
```
Comment (KateSakharova) [inline subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf:20]: I really like approach like
```
ch_dbs.multiMap { fasta, tax, otu, mcluster, label ->
        db_fasta: fasta
        db_tax: tax 
        db_otu: otu 
        db_mscluster: mscluster 
        db_label: label
    }.set {
        input
    }
```
and then use it like `MAPSEQ(ch_fasta, [input.db_fasta, input.db_tax, input.db_mscluster] )`
That multiMap is very convenient if you need to assign `meta` to each element in the input and create separate tuples. In your case is not relevant much, but usage of 0,1,2,3,4 might be confusing.

Actually, in that subworkflow makes sense to combine only:
```
ch_dbs.multiMap { fasta, tax, otu, mcluster, label ->
        mapseq_input: [fasta, tax, mscluster]
        mapseq_to_biom_input: [ otu, label ] 
    }.set {
        input
    }
```
and use `MAPSEQ(ch_fasta, input.mapseq_input)` and `MAPSEQTOBIOM(MAPSEQ.out.mseq, input.mapseq_to_biom_input)`

**Review by KateSakharova — APPROVED** [review]
Looks good!

**Review by Ales-ibt — APPROVED** [review]
It looks Ok to me :D

File: subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        MAPSEQ2BIOM.out.krona_input
+    )
+    ch_versions = ch_versions.mix(KRONA_KTIMPORTTEXT.out.versions.first())
+
+    emit:
+    mseq                  = MAPSEQ.out.mseq                   // channel: [ val(meta), [ mseq ] ]
+    krona_input           = MAPSEQ2BIOM.out.krona_input       // channel: [ val(meta), [ txt ] ]
+    biom_out              = MAPSEQ2BIOM.out.biom_out          // channel: [ val(meta), [ tsv ] ]
+    biom_notaxid_out      = MAPSEQ2BIOM.out.biom_notaxid_out  // channel: [ val(meta), [ tsv ] ]
+    html                  = KRONA_KTIMPORTTEXT.out.html       // channel: [ val(meta), [ bam ] ]
```
[the remark this answers is not in the document]
Comment (chrisAta) [reply subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf:44]: oh thank you good catch!!

File: subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf (hunk 1 above)
Comment (chrisAta) [reply subworkflows/ebi-metagenomics/mapseq_otu_krona/main.nf:20]: this is :hackerman: stuff!! I love it, making this change

**Pull request merged**